backend/logger/logger.py

```diff
@@ -0,0 +1,92 @@
+from fileHandler import FileHandler
+from singleton import Singleton
+
+from datetime import datetime
+import os
+from inspect import stack, getframeinfo
+
+
+class Logger(metaclass=Singleton):
+    """
+    Main class for logging functionalities.
+    """
+
+    def __init__(self, file_name="session", level=0):
+        self.level = level
+        fileHandler = FileHandler(file_name)
+        self.writer = fileHandler.getWriteHandle()
+
+    def debug(self, message):
+        """Write a debug message to the file using the writer
+
+        Parameters:
+        message (str) : The message you want to output.
+        """
+        if self.level > 0:
+            return
+        frameinfo = getframeinfo(stack()[1][0])
+        self.writer.write(
```

Comment:
please use f-strings, makes string concatenation much cleaner. 

The f-string counterpart for this would be 
`f"({str(date time.now().time()}) DEBUG: {frameinfo.lineno} : {message} \n"`
which is much more readable and clean imo

Reply:
Yes thank you for the reminder, haven't written python for a very very long time.

Reply:
np :D